physical/physical_test.go

```diff
@@ -151,43 +151,31 @@ func testBackend(t *testing.T, b Backend) {
 		t.Fatalf("missing child")
 	}
 
-	// Make a second nested entry to test prefix removal
```

Comment:
Hi,

This looks pretty good but one request: you changed the tests here but actually took out some functionality. The previous version was testing that removing one key from a prefix with two keys under it would not remove the prefix. You should add your test in addition to that, not instead of.